### ISSUE TITLE
Proposal: Add check to discourage redundant collection suffix (userList -> users)

### ISSUE THREAD

**Issue by shivam5285** [sevntu-checkstyle/sevntu.checkstyle#1155]
**Problem**:
Variables of type Collection are often named with redundant suffixes like
userList, orderSet, etc.

**Example**:
List<User> userList;

**Expected**:
List<User> users;

**Proposal**:
Introduce a new check that:
- Detects variables of type Collection/List/Set
- Flags names ending with List/Set/Collection
- Suggests plural naming instead

I am willing to implement this check and contribute a PR.

**Comment by romani** [comment]
After you finish, please share results, to help community to reuse solution if someone needs the same

**Comment by Sidhant2104** [comment]
Hi @romani, @shivam5285,

I’d like to work on this using MatchXpath as suggested. Since there hasn’t been an update for a while, I’ll proceed with an implementation and share the XPath along with test cases for review.

Please let me know if anyone is already actively working on it.

Thanks!

**Comment by romani** [comment]
@Sidhant2104 , please share xpath config to address this issue

**Comment by Sidhant2104** [comment]
Hi @romani,

I tried this locally using Checkstyle CLI.

It is correctly flagging:
- List<String> userList
- Set<Integer> orderSet

And it is not flagging:
- List<String> users
- int count

Here is the XPath config I used:

```xml
<module name="MatchXpath">
    <property name="query"
        value="//VARIABLE_DEF[
            ./TYPE//IDENT[
                @text='List' or
                @text='Set' or
                @text='Collection'
            ]
            and
            ./IDENT[
                matches(@text, '.*(List|Set|Collection)$')
            ]
        ]"/>
</module>
```

Also attaching screenshots of the test file and output.

Let me know if any changes are needed 

<img width="1489" height="214" alt="Image" src="https://github.com/user-attachments/assets/d52ba14b-f01b-46da-87ba-dec248a7b380" />
<img width="751" height="292" alt="Image" src="https://github.com/user-attachments/assets/d5565d59-e666-4eff-b7f4-a5bc39e1921b" />

**Comment by romani** [comment]
Please share all in text, input file, whole config, and whole CLI command and output

**Comment by romani** [comment]
@shivam5285, does this cover your request ?

**Comment by Sidhant2104** [comment]
Hi @romani,

Here is a complete reproducible example.

**Input file (Test.java):**
```java
import java.util.*;

class Test {
    List<String> userList;   // should be flagged
    Set<Integer> orderSet;   // should be flagged
    List<String> users;      // should not be flagged
    int count;               // should not be flagged
}
```

**Checkstyle configuration (checkstyle.xml):**
```xml
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="MatchXpath">
            <property name="query"
                value="//VARIABLE_DEF[
                    ./TYPE//IDENT[
                        @text='List' or
                        @text='Set' or
                        @text='Collection'
                    ]
                    and
                    ./IDENT[
                        matches(@text, '.*(List|Set|Collection)$')
                    ]
                ]"/>
        </module>
    </module>
</module>
```

**Command used:**
```bash
java -jar checkstyle.jar -c checkstyle.xml Test.java
```

**Output:**
```text
Starting audit...
[ERROR] Test.java:4:5: Illegal code structure detected. [MatchXpath]
[ERROR] Test.java:5:5: Illegal code structure detected. [MatchXpath]
Audit done.
Checkstyle ends with 2 errors.
```

Let me know if any adjustments are needed.

**Comment by romani** [comment]
I think it fully resolves issue.
Thanks a lot!!

I am closing issue, but discussion a can continue if required.

**Comment by Sidhant2104** [comment]
@romani Thanks for the review, glad it helped.

Happy to take this further if needed.
